src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/ExceptionHandler.kt

```diff
@@ -44,7 +46,14 @@ class HintExceptionHandler : ResponseEntityExceptionHandler() {
     @ExceptionHandler(PSQLException::class)
     protected fun handlePSQLException(e: PSQLException, request: HttpServletRequest): ResponseEntity<Any> {
         logger.error(e.message)
-        return ErrorDetail(HttpStatus.INTERNAL_SERVER_ERROR, "An unexpected error occurred.").toResponseEntity()
+        return ErrorDetail(HttpStatus.INTERNAL_SERVER_ERROR, "An unexpected error occurred.".appendErrorCodeInstructions())
+                .toResponseEntity()
+    }
+
+    private fun String.appendErrorCodeInstructions(): String {
+        val code = errorCodeGenerator.newCode()
+        val fullStopIfNeeded =  if (this.endsWith('.')) ""  else "."
+        return "$this$fullStopIfNeeded Please contact support at ${appProperties.supportEmail} and quote this code: $code"
```

Comment:
Shall we change this to: "If you see this message while you are using naomi a workshop, please contact your workshop technical support and show them this code. Otherwise <same as previous>"